### PR TITLE
Flip response-time-11 to use ZGC

### PR DESCRIPTION
**Pull request by j-baker**

### CODE REVIEW

File: gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/gc/GcProfile.java [hunk 1]
```diff
@@ -87,8 +88,9 @@ public final void newRatio(int newerRatio) {
         @Override
         public final List<String> gcJvmOpts() {
             return ImmutableList.of(
-                    // https://wiki.openjdk.java.net/display/shenandoah/Main
-                    "-XX:+UseShenandoahGC",
+                    // https://wiki.openjdk.java.net/display/zgc/Main
+                    "-XX:+UnlockExperimentalVMOptions",
+                    "-XX:+UseZGC",
```
Comment (CRogers) [inline gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/gc/GcProfile.java:93]: Why is this an improvement?

**Comment by carterkozak** [comment]
I don't believe zgc supports compressed oops, so heaps lower than 31.5g will effectively shrink.
@schlosna might have feedback on this change.

File: changelog/@unreleased/pr-664.v2.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+type: improvement
+improvement:
+  description: Flip response-time-11 to use ZGC
```
Comment (carterkozak) [inline changelog/@unreleased/pr-664.v2.yml:3]: If this goes through, we should note that this adds a new `response-time-11` gc profile.

Comment (robert3005) [reply changelog/@unreleased/pr-664.v2.yml:3]: We used to have this profile but removed it as it was relying on redhats jdk builds with shenandoahgc. The only confusing thing about ZGC is that it's linux only

Comment (carterkozak) [reply changelog/@unreleased/pr-664.v2.yml:3]: Can we put that in the changelog?

**Pull request closed without merging**